src/websockethandler.h

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef WEBSOCKETHANDLER_H
+#define WEBSOCKETHANDLER_H
+
+#include <QWebSocket>
+
+class QTimer;
+
+class WebSocketHandler final : public QObject {
+  Q_OBJECT
+ public:
+  WebSocketHandler();
+  void initialize();
+
+ signals:
+  void closed();
+
+ private:
+  QWebSocket m_webSocket;
+  QUrl m_url;
+  QTimer* m_pingTimer = nullptr;
+
+  bool m_reconnectOnClose = true;
+  bool m_waitingForPong = false;
+
+  void open();
+  void close();
+  void ping();
+  static QString webSocketServerUrl();
+  static bool isUserAuthenticated();
+
+  void onUserStateChanged();
+  void onConnected();
+  void onClose();
+  void onMessageReceived(const QString& message);
+  void onError(QAbstractSocket::SocketError error);
+  void onPong(quint64 elapsedTime);
```

Comment:
Usually our class headers are:

```
class Foo {
  public:
    Foo();

  signals:
    void signal();

  private:
    methods();

  private:
   bool m_var = true;
}
```

I would move the member variables at the bottom in their 'private' block.